### PR TITLE
Added Voici instructions (noting they are in active development)

### PR DESCRIPTION
**Pull request by JuanCab**
I had some challenges getting `voici build` running (and I don't believe it is producing proper output yet).  I have placed them online for now in this draft PR so you all can see if your luck is better than mine.

This branches off my current widgets tutorial (PR#8) which has not yet been merged with main..

### CODE REVIEW

**Comment by JuanCab** [comment]
I revised the `voici` instructions.  Note this required changing the `environment.yml` to add `micromamba` which is currently the only reliable build tool for voici.  Its a fairly short writeup, so I would appreciate any feedback.

**Review by mwcraig — CHANGES_REQUESTED** [review]
Only one important change: drop `widgets_pydantic.py` from the `dashboard` folder.

I know there are some typos in the updates, and it might make sense to rename the notebook so that it just covers voici, but those are changes @nicole-brewer and I can make while you are gone.

I will try out the install/build instructions in a few minutes.

**Comment by mwcraig** [comment]
Also, I'd like to get #8 in before we merge this or there will be a bunch of conflicts.

File: 04b_deployment_clientside.ipynb [hunk 1]
```diff
@@ -24,6 +24,90 @@
...
+    "cd voici_project\n",
+    "voici build --contents webapp"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d39a0660",
+   "metadata": {},
+   "source": [
+    "At this point you can test and see if the web application works by running a local web server pointed at the `_output` directory and then openning a new browser tab pointed at the local server [(http://localhost:8000/)](http://localhost:8000/).  If it works, you can deploy the `_output` directory to a web server. If you execute the following cell, you will need to \"Interrupt\" the kernel to continue with the rest of the notebook."
```
Comment (mwcraig) [inline 04b_deployment_clientside.ipynb]:
```suggestion
    "At this point you can test and see if the web application works by running a local web server pointed at the `_output` directory and then opening a new browser tab pointed at the local server [(**http://localhost:8000/**)](http://localhost:8000/).  If it works, you can deploy the `_output` directory to a web server. If you execute the following cell, you will need to \"Interrupt\" the kernel to continue with the rest of the notebook."
```

**Comment by mwcraig** [comment]
One last comment for now -- the fix in #8 in `key/dashboard/widgets.py` (which will get picked up once #8 is in and this is rebased) is necessary for the dashboard to work.

**Comment by JuanCab** [comment]
> One last comment for now -- the fix in #8 in `key/dashboard/widgets.py` (which will get picked up once #8 is in and this is rebased) is necessary for the dashboard to work.

Went ahead and rebased this branch to the current `widgets-tweaks` branch.

**Comment by mwcraig** [comment]
Can you try rebasing on main? If you prefer I can give that a go, still have about an hour of working time tonight until we get to our hotel.

**Comment by JuanCab** [comment]
> Can you try rebasing on main? If you prefer I can give that a go, still have about an hour of working time tonight until we get to our hotel.

Rebased onto current main (a lot of conflict resolution involved, bleh).

**Pull request closed without merging**